### PR TITLE
only aggr the statement when count is bigger than 1

### PR DESCRIPTION
**Pull request by gavinyue**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue # https://github.com/matrixorigin/matrixone/issues/12736
https://github.com/matrixorigin/matrixone/issues/12168

## What this PR does / why we need it:

do not aggr statement if aggr_count is smaller than 1

### CODE REVIEW

**Comment by matrix-meow** [comment]
@gavinyue Thanks for your contributions!

Review:

Title: only aggr the statement when count is bigger than 1

Body:
- This is a bug fix.
- It fixes issue #12736.
- The purpose of this PR is to not aggregate the statement if the aggr_count is smaller than 1.

Changes:
- In the `genETLData` function in the `buffer_pipe.go` file, the code has been modified.
- Previously, the statement was always aggregated if `stmt.AggrCount` was greater than 0.
- Now, the statement will only be aggregated if `stmt.AggrCount` is greater than 1.
- If `stmt.AggrCount` is 1 or less, the statement will not be aggregated.

Suggestions for improvement:
- The changes made in this pull request seem reasonable and address the bug mentioned in the issue.
- However, it would be helpful to have more context about the purpose of aggregating the statement and the significance of `stmt.AggrCount`.
- It would also be beneficial to have some test cases to ensure that the behavior is correct.
- Additionally, it would be good to add some comments explaining the purpose of the code and the reasoning behind the changes.
- Overall, the changes seem fine, but providing more context and adding comments and test cases would improve the quality and maintainability of the code.

**Review by xzxiong — COMMENTED** [review]
目标是：当只有一条记录时（即 `aggr_count == 1`），不要进行聚合的操作
- 不要清空或格式化字段的信息，例如：error、request_at, response_at 等等
- 保留原始的记录信息，即跟没触发聚合一样。

**Comment by gavinyue** [comment]
> 目标是：当只有一条记录时（即 `aggr_count == 1`），不要进行聚合的操作
> 
> * 不要清空或格式化字段的信息，例如：error、request_at, response_at 等等
> * 保留原始的记录信息，即跟没触发聚合一样。

已经更新， 所有的信息都是在有聚合的时候才会修改了

File: pkg/util/trace/impl/motrace/Aggr.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ type Aggregator struct {
 	Grouped     map[interface{}]Item
 	WindowSize  time.Duration
 	NewItemFunc func(i Item, ctx context.Context) Item
-	UpdateFunc  func(existing, new Item)
+	UpdateFunc  func(existing, new Item, ctx context.Context)
```
Comment (xzxiong) [inline pkg/util/trace/impl/motrace/Aggr.go:33]: go规范来说，如果有 `Context`参数，这个应该放在 第一

**Review by xzxiong — APPROVED** [review]

File: pkg/util/trace/impl/motrace/Aggr.go (hunk 1 above)
Comment (gavinyue) [reply pkg/util/trace/impl/motrace/Aggr.go]: 已经修改

**Review by heni02 — APPROVED** [review]

**Review by heni02 — APPROVED** [review]

**Review by aressu1985 — APPROVED** [review]

**Review by xzxiong — APPROVED** [review]

File: pkg/util/trace/impl/motrace/buffer_pipe.go [hunk 2]
```diff
@@ -191,10 +191,8 @@ func genETLData(ctx context.Context, in []IBuffer2SqlItem, buf *bytes.Buffer, fa
 
 			stmt, stmt_ok := i.(*StatementInfo)
 			if stmt_ok {
-				if stmt.AggrCount > 0 {
+				if stmt.AggrCount > 1 {
 					stmt.Statement = "/* " + strconv.FormatInt(stmt.AggrCount, 10) + " queries */ \n" + stmt.StmtBuilder.String()
-				} else {
-					stmt.Statement = stmt.StmtBuilder.String()
 				}
```
Comment (fengttt) [inline pkg/util/trace/impl/motrace/buffer_pipe.go:196]: What happens to the else branch?

Comment (gavinyue) [reply pkg/util/trace/impl/motrace/buffer_pipe.go:196]: 就是没有聚合， 它的Statement 就是原来的Statement

Comment (fengttt) [reply pkg/util/trace/impl/motrace/buffer_pipe.go:196]: Don't you need to do this line?
stmt.Statement = stmt.StmtBuilder.String()

Comment (gavinyue) [reply pkg/util/trace/impl/motrace/buffer_pipe.go:196]: 如果是只有一条的话, statement 里面的内容和stmt builder 生成是一样的。 所以不需要。

**Pull request merged**